src/e2e/create_pool.rs

```diff
@@ -168,4 +177,142 @@ pub mod e2e_tests {
 
         Ok(())
     }
+
+    #[ink_e2e::test]
+    async fn test_create_pool_init_sqrt_price_minimal_difference_from_tick(
+        mut client: ink_e2e::Client<C, E>,
+    ) -> E2EResult<()> {
+        let dex = create_dex!(client, ContractRef, Percentage::new(0));
+        let (token_x, token_y) = create_tokens!(client, TokenRef, 500, 500);
+        let alice = ink_e2e::alice();
+
+        let fee_tier = FeeTier::new(Percentage::from_scale(5, 1), 3).unwrap();
+        let init_tick = 0;
+        let init_sqrt_price = calculate_sqrt_price(init_tick).unwrap() - SqrtPrice::new(1);
+        add_fee_tier!(client, ContractRef, dex, fee_tier, alice).unwrap();
+
+        create_pool!(
+            client,
+            ContractRef,
+            dex,
+            token_x,
+            token_y,
+            fee_tier,
+            init_sqrt_price,
+            init_tick,
+            alice
+        )
+        .unwrap();
+
+        let pool = get_pool!(client, ContractRef, dex, token_x, token_y, fee_tier).unwrap();
+        assert_eq!(
+            pool.current_tick_index,
+            init_tick - fee_tier.tick_spacing as i32
+        );
+
+        Ok(())
+    }
+
+    #[ink_e2e::test]
+    async fn test_create_pool_init_sqrt_price_has_closer_init_tick(
+        mut client: ink_e2e::Client<C, E>,
+    ) -> E2EResult<()> {
+        let dex = create_dex!(client, ContractRef, Percentage::new(0));
+        let (token_x, token_y) = create_tokens!(client, TokenRef, 500, 500);
+        let alice = ink_e2e::alice();
+
+        let fee_tier = FeeTier::new(Percentage::from_scale(5, 1), 1).unwrap();
+        let init_tick = 2;
+        // tick = 3 -> 1.000150003749000000000000
+        // between  -> 1.000175003749000000000000
+        // tick = 4 -> 1.000200010000000000000000
+        let init_sqrt_price = SqrtPrice::new(1000175003749000000000000);
+        add_fee_tier!(client, ContractRef, dex, fee_tier, alice).unwrap();
+
+        let result = create_pool!(
+            client,
+            ContractRef,
+            dex,
+            token_x,
+            token_y,
+            fee_tier,
+            init_sqrt_price,
+            init_tick,
+            alice
+        );
+        assert_eq!(result, Err(InvariantError::InvalidInitSqrtPrice));
+        let correct_init_tick = 4;
+        create_pool!(
+            client,
+            ContractRef,
+            dex,
+            token_x,
+            token_y,
+            fee_tier,
+            init_sqrt_price,
+            correct_init_tick,
+            alice
+        )
+        .unwrap();
+
+        let pool = get_pool!(client, ContractRef, dex, token_x, token_y, fee_tier).unwrap();
+        assert_eq!(
+            pool.current_tick_index,
+            correct_init_tick - fee_tier.tick_spacing as i32
```

Comment:
If you need to modify `current_tick` it isn't correct tick. So there is definitely problem with presenting this test.
correct tick for this sqrt_price: `1000175003749000000000000` is `3`